proxy/src/http/control.rs

```diff
@@ -163,13 +177,25 @@ mod handler {
     }
 
     /// Reset the coco state by creating a new temporary directory for the librad paths.
-    pub async fn nuke_coco(librad_paths: Arc<RwLock<Paths>>) -> Result<impl Reply, Rejection> {
-        let dir = tempfile::tempdir().expect("tmp dir creation failed");
-        let new = Paths::from_root(dir.path()).expect("unable to get paths");
+    pub async fn nuke_coco(peer: Arc<Mutex<coco::Peer>>) -> Result<impl Reply, Rejection> {
+        // TmpDir deletes the temporary directory once it DROPS.
+        // This means our new directory goes missing, and future calls will fail.
+        // The Peer creates the directory again.
+        //
+        // N.B. this may gather lot's of tmp files on your system. We're sorry.
+        let tmp_path = {
+            let temp_dir = tempfile::tempdir().expect("test dir creation failed");
+            temp_dir.path().to_path_buf()
+        };
 
-        let mut paths = librad_paths.write().await;
+        let config = coco::default_config(
+            SecretKey::new(),
+            tmp_path.to_str().expect("path extraction failed"),
+        )?;
+        let new_peer = coco::Peer::new(config).await?;
```

Comment:
Something I'm worried about is when we start handling the run loop in peer. How will this work with nuking o_O

Reply:
CHANNELS!